Rustls support with dynamic choice of TLS provider for cloud (#1254)
scylladb/scylla-rust-driver: docs/source/connecting/connecting.md

Author: Well, we did not even have any non-cloud tests for openssl. In TLS workflow we just run the example, where we connect to the cluster using openssl. Analogous example is introduced and run in CI for rustls.

use std::path::Path;
use std::error::Error;
use scylla::client::session_builder::CloudSessionBuilder;
use scylla::cloud::CloudTlsProvider;

#[tokio::main]
async fn main() -> Result<(), Box<dyn Error>> {
    let session =
        CloudSessionBuilder::new(Path::new("config_data.yaml"), CloudTlsProvider::OpenSsl010)
            .unwrap()
            .build()
            .await
            .unwrap();

    Ok(())
}
# }
```

> ***Note***\
> `CloudSessionBuilder::new()` accepts two parameters. One of them is path to the configuration file.

> > Second parameter is responsible for choosing the underlying TLS provider library.

Reviewer: :wrench:

Reviewer: ```suggestion
> `CloudSessionBuilder::new()` accepts two parameters. The first is a path to the configuration file.
> The second parameter is responsible for choosing the underlying TLS provider library.
```
